crates/analytics/docs/clickhouse/scripts/sdk_events.sql

```diff
@@ -0,0 +1,232 @@
+    `platform` LowCardinality(Nullable(String)),
+    `source` LowCardinality(Nullable(String)),
+    `category` LowCardinality(Nullable(String)),
+    `version` LowCardinality(Nullable(String)),
+    `value` Nullable(String),
+    `component` LowCardinality(Nullable(String)),
+    `payment_method` LowCardinality(Nullable(String)),
+    `payment_experience` LowCardinality(Nullable(String)) DEFAULT '',
+    `created_at` DateTime64(3) DEFAULT now64() CODEC(T64, LZ4),
+    `inserted_at`  DateTime DEFAULT now() CODEC(T64, LZ4), 
+    `latency` Nullable(UInt32) DEFAULT 0,
+) ENGINE = MergeTree PARTITION BY merchant_id
+ORDER BY
+    (merchant_id, payment_id)
+    TTL inserted_at + toIntervalMonth(18)
+SETTINGS index_granularity = 8192;
+
+CREATE MATERIALIZED VIEW sdk_events_parse_errors (
+    `topic` String,
+    `partition` Int64,
+    `offset` Int64,
+    `raw` String,
+    `error` String
+) ENGINE = MergeTree
+ORDER BY
+    (topic, partition, offset) SETTINGS index_granularity = 8192 AS
+SELECT
+    _topic AS topic,
+    _partition AS partition,
+    _offset AS offset,
+    _raw_message AS raw,
+    _error AS error
+FROM
+    sdk_events_queue
+WHERE
+    length(_error) > 0;
+
+CREATE MATERIALIZED VIEW sdk_events_audit_mv TO sdk_events_audit (
+    `payment_id` Nullable(String),
+    `merchant_id` String,
+    `remote_ip` Nullable(String),
+    `log_type` LowCardinality(Nullable(String)),
+    `event_name` LowCardinality(Nullable(String)),
+    `first_event` Bool,
+    `latency` Nullable(UInt32),
+    `browser_name` LowCardinality(Nullable(String)),
+    `browser_version` Nullable(String),
+    `platform` LowCardinality(Nullable(String)),
+    `source` LowCardinality(Nullable(String)),
+    `category` LowCardinality(Nullable(String)),
+    `version` LowCardinality(Nullable(String)),
+    `value` Nullable(String),
+    `component` LowCardinality(Nullable(String)),
+    `payment_method` LowCardinality(Nullable(String)),
+    `payment_experience` LowCardinality(Nullable(String)),
+    `created_at` DateTime64(3),
+    `created_at_precise` DateTime64(3),
+    `inserted_at` DateTime DEFAULT now() CODEC(T64, LZ4)
+) AS
+SELECT
+    payment_id,
+    merchant_id,
+    remote_ip,
+    log_type,
+    event_name,
+    multiIf(first_event = 'true', 1, 0) AS first_event,
+    latency,
+    browser_name,
+    browser_version,
+    platform,
+    source,
+    category,
+    version,
+    value,
+    component,
+    payment_method,
+    payment_experience,
+    toDateTime64(timestamp, 3) AS created_at,
+    toDateTime64(timestamp, 3) AS created_at_precise,
+    now() AS inserted_at
+FROM
+    sdk_events_queue
+WHERE
+    (length(_error) = 0)
+    AND (payment_id IS NOT NULL);
+
+CREATE TABLE sdk_active_payments ( 
+    `payment_id` Nullable(String),
+    `merchant_id` String,
+    `created_at` DateTime,
+    INDEX merchantIndex merchant_id TYPE bloom_filter GRANULARITY 1
+) ENGINE = ReplacingMergeTree
+PARTITION BY toStartOfSecond(created_at)
```

Comment:
is this needed? too many partitions will be created

Reply:
Yes. There would be 60 active partitions at a given point of time. Rest will be inactive